### ISSUE TITLE
PubCenter - No ad available

### ISSUE THREAD

**Issue by negue** [Adrotator/AdrotatorV2#34]
(Sorry I have to create a new issue, but I couldn't post on #26, GitHub won't let me^^)

I have a problem with PubCenter, sometimes I get an Ad and sometimes I dont, but it seems that the Task isn't awaited and so I get this Exception.

```
28|2014-10-06T22:57:27.0106241+00:00|ERROR|2|App|UnobservedTaskException --> System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> MicrosoftAdvertising.Shared.AdException: No ad available.
   at MicrosoftAdvertising.Shared.AdPlacement.<SendAdRequestAsync>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MicrosoftAdvertising.Shared.AdPlacement.<RefreshAsync>d__1c.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) MicrosoftAdvertising.Shared.AdException: No ad available.
   at MicrosoftAdvertising.Shared.AdPlacement.<SendAdRequestAsync>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MicrosoftAdvertising.Shared.AdPlacement.<RefreshAsync>d__1c.MoveNext()<---

```

I know it is possible that there is "no ad available" but then AdRotator shoud switch to the next one, but currently it seems that It doesn't know that there was an error^^

**Comment by SimonDarksideJ** [comment]
This is a known issue with PubCenter and one we keep trying to tackle.
The problem is that in one of the recent releases, PubCenter changed from error events to exceptions, which is much harder to manage.  AdRotator should still capture the "No Ad" exception and treat it as a failure then move on to the next provider.  If you find it isn't doing so, then let us know and we'll investigate.

**Comment by negue** [comment]
I'll try to create a simple app and see if the problem is still there. 

But currently in my app I added 2 adrotatorcontrols in one page, sometimes only 1 of em shows something the other does nothing (but should switch to AdDuplex)

**Comment by SimonDarksideJ** [comment]
Ok, we have an update coming soon (very soon now thanks to one contributor) for Windows 8.1/WP8.1 /Universal apps.
So will ensure we try and test this scenario further.

**Comment by SimonDarksideJ** [comment]
Sadly we could not subscribe to ad failures for PubCenter, the move to WinRT components hampers us greatly.
Might try and find a workaround in the next release. 

For now, try using a shorter AdRefresh time or the sliding ads functionality
